### PR TITLE
backward compatibility of pointwise activations

### PR DESCRIPTION
**Pull request by szagoruyko**
small fix for #19

### CODE REVIEW

File: ReLU.lua [hunk 1]
```diff
@@ -4,3 +4,8 @@ function ReLU:__init(inplace)
    parent.__init(self, inplace)
    self.mode = 'CUDNN_ACTIVATION_RELU'
 end
+
+function ReLU:updateOutput(input)
+  if not self.mode then self.mode = 'CUDNN_ACTIVATION_RELU' end
```
Comment (soumith) [inline ReLU.lua]: self.mode = self.mode or 'CUDNN_ACTIVATION_RELU'
is nicer to read.

**Comment by soumith** [comment]
i knew this was going to happen when I pushed the change, but cudnn was not being used by many back then, and the refactor HAD to be done as a cleanup, so pushed it.

**Comment by soumith** [comment]
when in doubt about something breaking, please feel free to ask, I usually know the answer about serialization breakage...especially if I broke it myself.

in nn/cunn i painfully make sure that the serialization is ALWAYS backward compatible, and cudnn will also be kept that way going forward.

however, breakage will occur when we move to R2 bindings again. I will create a tagged branch called R1, that can be used in that case.

**Comment by szagoruyko** [comment]
thanks Soumith, will keep this in mind

**Comment by szagoruyko** [comment]
did you mean that breakage will be caused by the same reason that this PR or you had something else in mind?
I've tested CUDNN v2 on ubuntu and os x, seems to work well, so I will add groups option to R2v2 and it lacks in-place pointwise activations, right?

**Comment by soumith** [comment]
https://github.com/soumith/cudnn.torch/commits/R2RC2
already added groups, and in-place

**Comment by szagoruyko** [comment]
wow, missed that, cool!
I've tried to run forward in v2 with model saved in with v1, segfaults.

**Comment by soumith** [comment]
sigh!

**Comment by szagoruyko** [comment]
working on it, fixed the segfault (it was padW and padH in Pooling), applying this PR

**Pull request merged**